### PR TITLE
Add stale issues bot to help triage efforts

### LINKED ISSUE #28900: Stale Issue Automation
Thanks to everyone who provided feedback and contributed to the discussion around [the proposal posted on Make here](https://make.wordpress.org/core/2021/01/14/stale-issues-in-gutenberg-repository/). We’ve listened to all the ideas there and are proposing the following based on what was discussed:

* The bot will not close any issues. 

* After 6 months of no activity, the bot will help automatically label issues with `[Status] Needs confirmation`. If no action is taken after 1 month with this new label, the bot will then add `[Status] Stale`. This should help those triaging issues. 
 
* If any comment is made after the actions taken by the bot the labels added will be removed. 

The goal of the bot is to re-engage users by prompting them to confirm the problem is still valid with the latest versions.

For context, this is the proposed working message that the bot will apply as a comment after adding the `[Status] Needs confirmation` label to the issue itself:

> Hi @username,
> This ticket has gone 180 days without any activity and therefore meets the project’s requirement for a check-in to ensure it is not obsolete. If you are still experiencing this issue, then please reply with confirmation and any updated reproduction steps.
> Thanks,
> GutenBot

Related, in a parallel issue, we implemented a bot to enforce the Needs Info policy. The policy has been in place for a while, [see triage document](https://github.com/WordPress/gutenberg/blob/master/docs/contributors/triage.md), but the bot to enforce was new ([see PR](https://github.com/WordPress/gutenberg/pull/28179)). I wanted to call it out after all the feedback on the post, it seems like we might want to adjust that policy and not have a bot enforce it.

### PR DESCRIPTION
**Pull request by mkaz**
## Description

Closes #28900

- Adds two stale bots to help triage efforts by labeling issues

1. After 180 days of no activity, a reminder message will be posted to
the issue and the "Needs Testing" label applied. This bot does not
remove the label automatically after a comment, since there is no way to
know if the comment is actually related to the testing.

2. After 30 days of no activity on an issue with the "Needs Testing"
label the issue will be marked "[Status] Stale" with no message posted.
If a comment is made, the stale label is removed.

## How has this been tested?

Not an easy one to test since it is uses GitHub workflow.

The best bet is to review the parameters in the PR and the [documentation for the stale bot used](https://github.com/actions/stale).

## Types of changes

Adds two new GitHub actions, one for marking items as Needs Testing, and the second to mark items that need testing as Stale. As explained in the definition above.

### CODE REVIEW

File: .github/workflows/stale-issue-add-needs-testing.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/stale@v3
+      with:
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
+        stale-issue-message: "Hi,\nThis issue has gone 180 days without any activity. This means it's time for a check-in to make sure it is relevant today. If you are still experiencing this issue, you can help the project by responding to confirm the problem and by providing any updated reproduction steps.\nThanks for helping out."
+        days-before-stale: 180
```
Comment (paaljoachim) [inline .github/workflows/stale-issue-add-needs-testing.yml:14]: Hi,\nThis issue has gone 180 days without any activity. This means it's time for a check-in to make sure it is relevant today. 

Rephrasing.
Hi, this issue has gone 180 days without any activity. This means it is time for a check-in to make sure it is still relevant.

**Comment by paaljoachim** [comment]
The rest looks fine to me.
I am thinking that @annezazu can also take a quick look over. To see if there is anything that stands out that need adjusting.

File: .github/workflows/stale-issue-add-needs-testing.yml (hunk 1 above)
Comment (mkaz) [reply .github/workflows/stale-issue-add-needs-testing.yml:14]: Thanks! Updated.

File: .github/workflows/stale-issue-add-needs-testing.yml [hunk 2]
```diff
@@ -10,7 +10,7 @@ jobs:
     - uses: actions/stale@v3
       with:
         repo-token: ${{ secrets.GITHUB_TOKEN }}
-        stale-issue-message: "Hi,\nThis issue has gone 180 days without any activity. This means it's time for a check-in to make sure it is relevant today. If you are still experiencing this issue, you can help the project by responding to confirm the problem and by providing any updated reproduction steps.\nThanks for helping out."
+        stale-issue-message: "Hi,\nThis issue has gone 180 days without any activity. This means it is time for a check-in to make sure it is still relevant. If you are still experiencing this issue, you can help the project by responding to confirm the problem and by providing any updated reproduction steps.\nThanks for helping out."
```
Comment (annezazu) [inline .github/workflows/stale-issue-add-needs-testing.yml:13]: I might add in: "If you are still experiencing the issue with the latest versions" (to keep it short leaving off "versions of Gutenberg and WordPress").

Comment (mkaz) [reply .github/workflows/stale-issue-add-needs-testing.yml]: :+1: Added

**Review by annezazu — APPROVED** [review]
Ship it!

**Review by paaljoachim — APPROVED** [review]
Looks good!

**Pull request merged**